www/data_service/src/services/stream/stream.service.coffee

```diff
@@ -0,0 +1,36 @@
+class Stream extends Factory
+    constructor: ($log) ->
+        return class StreamInstance
+            # the unsubscribe listener will be called on each unsubscribe call
+            onUnsubscribe: null
+
+            constructor: ->
+                @listeners = []
+
+            subscribe: (listener) ->
+                if not angular.isFunction(listener)
+                    $log.debug("#{listener} is not a function")
```

Comment:
okay, I will throw TypeErrors in such cases

On Fri, Jun 12, 2015, 17:17 Chase Zhang <notifications@github.com> wrote:

> In www/data_service/src/services/stream/stream.service.coffee
> <https://github.com/buildbot/buildbot/pull/1703#discussion_r32324673>:
>
> > @@ -0,0 +1,36 @@
> > +class Stream extends Factory
> > +    constructor: ($log) ->
> > +        return class StreamInstance
> > +            # the unsubscribe listener will be called on each unsubscribe call
> > +            onUnsubscribe: null
> > +
> > +            constructor: ->
> > +                @listeners = []
> > +
> > +            subscribe: (listener) ->
> > +                if not angular.isFunction(listener)
> > +                    $log.debug("#{listener} is not a function")
>
> Maybe we should throw an error here? This interface is likely to be used
> only by ourselves, we may not let it pass silently(though there is a log).
> Instead an error stop the program as soon as possible is better.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/buildbot/buildbot/pull/1703/files#r32324673>.
>